### ISSUE TITLE
Unexpected labels inference using multi-class-model

### ISSUE THREAD

**Issue by original-doc** [LiamLian0727/USIS10K#6]
![train_00001](https://github.com/user-attachments/assets/fa914768-d684-40dc-8185-da14f8c2c9ef)
Thank you for your wonderful work!
The issue I meet with is that when I use the multiclass model to infer the segmentation,  it seems label all the marines with “bicycles” and "cars". I check the metadata of multi-class in the dataset.py, but to find there are no labels like these. Is this problem due to the ViT's pretraining on ImageNet and not get finetuned on this custom dataset? The test performance is still good, but the labels are quite confusing. 

```
@DATASETS.register_module()
class MultiClassUSIS10KInsSegDataset(CocoDataset):
    """Dataset for Cityscapes."""

    METAINFO = {
        'classes': ['wrecks/ruins', 'fish', 'reefs', 'aquatic plants',
                    'human divers', 'robots', 'sea-floor'],
        'palette': [(220, 20, 60), (255, 0, 0), (0, 0, 142), (0, 0, 70),
                    (0, 60, 100), (0, 80, 100), (0, 0, 230)]
    }
```
The checkpoints and USIS10K dataset are downloaded from the given links.

Test result:
```
coco/bbox_mAP: 0.4150  coco/bbox_mAP_50: 0.5970  coco/bbox_mAP_75: 0.4760  coco/bbox_mAP_s: 0.4440  coco/bbox_mAP_m: 0.3580  coco/bbox_mAP_l: 0.4510  coco/segm_mAP: 0.4390  coco/segm_mAP_50: 0.5960  coco/segm_mAP_75: 0.5000  coco/segm_mAP_s: 0.2860  coco/segm_mAP_m: 0.3370  coco/segm_mAP_l: 0.4910  data_time: 0.0135  time: 1.1944
```

**Comment by LiamLian0727** [comment]
I guess this seems to be caused by the use of MS COCO's labeled metadata, can you provide more detailed runtime information and config files and let me help you?

**Comment by LiamLian0727** [comment]
One possible solution is to manually specify the label list in the config file to print the correct category name at test time.

```
CLASSES = ('wrecks/ruins', 'fish', 'reefs', 'aquatic plants', 'human divers', 'robots', 'sea-floor')

train_dataloader = dict(
    batch_size=batch_size,
    num_workers=num_workers,
    persistent_workers=persistent_workers,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        indices=indices,
        data_root=data_root,
        metainfo=CLASSES,
        ann_file='multi_class_annotations/multi_class_train_annotations.json',
        data_prefix=dict(img='train'),
        # filter_cfg=dict(filter_empty_gt=True, min_size=32),
        pipeline=train_pipeline,
        backend_args=backend_args)
)

val_dataloader = dict(
    batch_size=batch_size,
    num_workers=num_workers,
    persistent_workers=persistent_workers,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        indices=indices,
        data_root=data_root,
        metainfo=CLASSES,
        ann_file='multi_class_annotations/multi_class_val_annotations.json',
        data_prefix=dict(img='val'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args)
)

test_dataloader = dict(
    batch_size=batch_size,
    num_workers=num_workers,
    persistent_workers=persistent_workers,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        indices=indices,
        data_root=data_root,
        metainfo=CLASSES,
        ann_file='multi_class_annotations/multi_class_test_annotations.json',
        data_prefix=dict(img='test'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args)
)
```

**Comment by original-doc** [comment]
The error was triggered by `UserWarning: dataset_meta or class names are not saved in the checkpoint's meta data, use COCO classes by default.`, it sconcerns with the provided checkpoints.

Solely changing the metainfo in dataloader didn't really help, I also added the meta to the checkpoints to solve this problem.

**Comment by original-doc** [comment]
> The error was triggered by `UserWarning: dataset_meta or class names are not saved in the checkpoint's meta data, use COCO classes by default.`, it sconcerns with the provided checkpoints.
> 
> Solely changing the metainfo in dataloader didn't really help, I also added the meta to the checkpoints to solve this problem.

also the class in config should be set as dict `CLASSES = dict(classes=['wrecks/ruins', 'fish', 'reefs', 'aquatic plants', 'human divers', 'robots', 'sea-floor'])`

**Comment by original-doc** [comment]
If you'd like to, I'm happy to pull the scripts for fixing this issue and inference to make this repo better. @LiamLian0727

**Comment by LiamLian0727** [comment]
Looking forward to your scripts and thanks for making our repository better!

**Comment by LiamLian0727** [comment]
I've merged your pull request into the main branch, thanks again for contributing to this repository!
